zanata-war/src/main/java/org/zanata/webtrans/server/TranslationWorkspaceManagerImpl.java

```diff
@@ -172,26 +177,47 @@ protected String getSessionId() {
     }
 
     @Observer(ProjectUpdate.EVENT_NAME)
-    public void projectUpdate(@Observes ProjectUpdate payload) {
-        projectUpdate(payload.getProject());
+    // transaction has already been committed and marked as rolled back only for
+    // current thread. We have to open a new transaction to load any lazy
+    // properties (otherwise exception like javax.resource.ResourceException:
+    // IJ000460: Error checking for a transaction: Transactions are not active)
+    @Async
```

Comment:
I thought that just by using the `Work` class you could do it in a new transaction. Not sure that you need a new thread for it to happen.

Also, I think you are able to fire events to be executed asynchronously (i.e new thread).